core/src/test/java/org/elasticsearch/threadpool/ScalingThreadPoolTests.java

```diff
@@ -177,7 +177,8 @@ public void testScalingThreadPoolThreadsAreTerminatedAfterKeepAlive() throws Int
                     }
                 });
             }
-            assertThat(stats(threadPool, threadPoolName).getThreads(), equalTo(128));
+            int threads = stats(threadPool, threadPoolName).getThreads();
+            assertEquals(128, threads);
```

Comment:
You are free to open a bug in Oracle's bug tracker! You can also investigate what they already have. But as this is not reproducible in isolation, I don't think you have a chance to fix this. Better to kill `assertThat` brokenness. I like it that Java 9 complains automatically about that stuff :-)